lib/Serge/Engine/Plugin/parse_yaml.pm

```diff
@@ -134,12 +141,30 @@ sub parse {
         die $error_text;
     }
 
+    if ($self->{data}->{yaml_mode} eq 'rails') {
+        my @tree_keys = keys(%$tree);
+        my $tree_count = int(@tree_keys);
+        if ($tree_count == 0) {
+            # Special case NOOP. Empty data
+        } elsif ($tree_count == 1) {
+            $tree = $tree->{$tree_keys[0]};
+        } else {
+            my $err = join(', ', @tree_keys);
+            $err = "YAML file with more than 1 key at root in yaml_mode: rails: $err";
+            $self->{errors}->{$self->{parent}->{engine}->{current_file_rel}} = $err;
+            # Or should we die?
+            warn "WARNING: $err";
+        }
+    }
+
     # Process tree recursively
 
     $self->process_node('', $tree, $callbackref, $lang);
 
     # Reconstruct YAML
 
+    $tree = {$lang => $tree} if ($lang && $self->{data}->{yaml_mode} eq 'rails');
```

Comment:
I also just realized that it’s not as simple as “Rails drops the region when it matches the language” because it uses `ja` in favor of `ja-JP`. 

There are probably other cases of which I am not aware. I may lean toward a Util method to capture this.